### PR TITLE
[IMP] udes_security: Add flag to restrict a user to a single session

### PR DESCRIPTION
**Pull request by joe-gb**
u_restrict_to_single_session flag added which restricts a user to a
single session. Upon login all active sessions for a user are deleted.
(story/9772)

### CODE REVIEW

File: addons/udes_security/models/res_users.py [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
 
+    u_restrict_to_single_session = fields.Boolean(
+        string="Whether a user is restricted to having a single login session.",
+        default=False
+    )
+
+    @classmethod
+    def authenticate(cls, db, *args, **kwargs):
+        """ Override to clear all previous sessions if authenticated """
+        sessions = {sid: _store.get(sid) for sid in _store.list()}
```
Comment (kenvac) [inline addons/udes_security/models/res_users.py]: Can we move this inside if condition?

Comment (joe-gb) [reply addons/udes_security/models/res_users.py]: thank you! updated :)

File: addons/udes_security/tests/test_single_session.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    def restrict_multiple_sessions(self, restrict):
+        with self.cursor() as cr:
+            env = self.env(cr)
+            env["res.users"].browse(self.user_1.id).u_restrict_to_single_session = restrict
+
+    def assertSessionValid(self, sids):
+        store_list = _store.list()
+        for sid in sids:
+            self.assertTrue((sid in store_list))
```
Comment (nocturnalastro) [inline addons/udes_security/tests/test_single_session.py]: why double brackets?

Comment (joe-gb) [reply addons/udes_security/tests/test_single_session.py]: good spot, fixed

**Pull request merged**